### PR TITLE
feat: allow opt-out of exporting environment variables

### PR DESCRIPTION
**Pull request by sethvargo**
This allows users to opt-out of exporting environment variables. This might be helpful if they **don't** want future steps to be authenticated, or if the exported environment variables conflict with other values.

Fixes https://github.com/google-github-actions/auth/pull/156

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -163,26 +164,52 @@ async function run(): Promise<void> {
...
+        'GCP_PROJECT',
+        'GOOGLE_CLOUD_PROJECT',
+      ].forEach((name) => {
+        const existing = process.env[name];
+        if (existing) {
+          const old = JSON.stringify(existing);
+          logWarning(`Overwriting existing environment variable ${name} (was: ${old})`);
+        }
+
+        exportVariable(name, computedProjectID);
```
Comment (bharathkkb) [inline src/main.ts:210]: Could we pull this into a function?

File: README.md [hunk 2]
```diff
@@ -209,6 +209,26 @@ regardless of the authentication mechanism.
         - uses: 'google-github-actions/auth@v0'
      ```
 
+-   `export_environment_variables`: (Optional) If true, the action will export
```
Comment (bharathkkb) [inline README.md:212]: Should we call out this defaults to true?

Comment (sethvargo) [reply README.md:212]: Done.

File: src/main.ts (hunk 1 above)
Comment (sethvargo) [reply src/main.ts]: Done.

**Review by bharathkkb — APPROVED** [review]

**Comment by simi** [comment]
:muscle: :partying_face:

**Pull request merged**